### PR TITLE
Add https example

### PR DESCRIPTION
**Pull request by talex5**
[ Making a PR now so that @avsm can link it in his slides - this is not ready yet ]

**- What I did**

I added a sample Mirage project that runs an Irmin-backed https web-server using processes communicating over capnp-rpc. This is intended to demonstrate the RPC system and Irmin.

**- How to verify it**

There is a README in the project directory with instructions. You can build and test everything using Docker, or build it on your host using opam.

**- Description for the changelog**

Add sample Mirage HTTPS service.

### CODE REVIEW

**Comment by avsm** [comment]
Thanks, I think this needs to be under the projects/miragesdk namespace for the moment.

**Comment by rneugeba** [comment]
not directly related, but the DHCP unikernel is under `./projects/miragesdk` which this code is under a different project. Could we either have all unikernel type work under one project or have the projects named better?

**Comment by avsm** [comment]
@rneugeba yes this should be under projects/miragesdk alongside dhcp for the moment. Just opened this PR for the Moby Security SIG in 45 minutes to make it easier to show the WIP :-)

File: projects/miragesdk/examples/https-unikernel/README.md [hunk 1]
```diff
@@ -0,0 +1,70 @@
+## https-unikernel example service
+
+This sample service implements an https web-server as a set of three components communicating using [Cap'n Proto RPC][].
```
Comment (AkihiroSuda) [inline projects/miragesdk/examples/https-unikernel/README.md]: hyperlink seems broken

File: projects/miragesdk/examples/https-unikernel/README.md [hunk 2]
```diff
@@ -0,0 +1,70 @@
+## https-unikernel example service
```
Comment (AkihiroSuda) [inline projects/miragesdk/examples/https-unikernel/README.md]: Does this already work as solo5 "unikernel"?
If not, it might be better to mention that?

Comment (talex5) [inline projects/miragesdk/examples/https-unikernel/README.md]: No, but getting all or some parts of it run as unikernels is the goal.

File: projects/miragesdk/examples/https-unikernel/README.md (hunk 1 above)
Comment (talex5) [inline projects/miragesdk/examples/https-unikernel/README.md]: Works for me.

Comment (AkihiroSuda) [inline projects/miragesdk/examples/https-unikernel/README.md]: np, my misread :sweat_smile:

**Comment by justincormack** [comment]
So what is the status of this?

**Comment by talex5** [comment]
Do "tracking" PRs need a status? Would be design-review if anything, I guess.

**Comment by justincormack** [comment]
If its not ready I would rather have an issue pointing to a branch. But its a project so we can merge stuff whenever, but obviously can cant force push after that...

**Comment by rneugeba** [comment]
If this get's merged, I'd prefer it to be part of `projects/miragesdk`, as pointed out above

**Comment by justincormack** [comment]
@rneugeba it seems to be under `projects/miragesdk` now...

**Comment by rneugeba** [comment]
ah missed that. And yes, I don't like "tracking" PRs, they are distracting either merge or keep a branch somewhere

**Comment by talex5** [comment]
I'll let @avsm decide what to do with this, as he was the one who asked for it.

**Comment by rn** [comment]
@avsm @talex5 asked for you view above if we should merge this or not.

trying to decide if to close or merge. Not a big fan of RFC PRs which are lingering for too long :)

**Comment by talex5** [comment]
Well, feel free to merge if you want then. It won't do any harm and it might be a useful example of doing RPC.

**Pull request merged**